gradle/hacks/global-exclude-dependencies.gradle

```diff
@@ -40,6 +40,7 @@ allprojects { prj ->
   ]
   configurations.matching { it.name in configNames }.all {
     exclude group: 'log4j', module: 'log4j' // use SLF4j
+    exclude group: "ch.qos.logback" // use SLF4j
     exclude group: 'commons-logging', module: 'commons-logging' // use SLF4j
     exclude group: 'com.google.code.findbugs', module: 'jsr305' // Uses GPL license
     exclude group: 'com.google.code.findbugs', module: 'annotations' // Use Spotbugs Annotations as replacement
```

Comment:
Just FYI (not specific to this PR), if you look at our generated POMs, you will see that each dependency has an excludes list of the list above.  This is even if what's excluded isn't even a transitive dependency to where it's attached.  Our system works but it's annoyingly verbose.

Reply:
I had this logback exclude under all the zookeeper stuff - but then logback came under curator and it just snowballed. It made way more sense to add a global exclude.

There MIGHT be a way to do this with some variables in excludes. That would be good to look at separately.

Reply:
The Gradle documentation describes a few approaches we should explore: https://docs.gradle.org/current/userguide/dependency_downgrade_and_exclude.html